include/podio/GenericParameters.h

```diff
@@ -243,5 +248,17 @@ std::vector<std::string> GenericParameters::getKeys() const {
   return keys;
 }
 
+template <typename T, typename>
+std::vector<std::vector<T>> GenericParameters::getValues() const {
+  std::vector<std::vector<T>> values;
+  {
+    auto& mtx = getMutex<T>();
+    const auto& map = getMap<T>();
+    std::lock_guard lock{mtx};
```

Comment:
Whenever locks are being used one should document where why and when they are needed